### PR TITLE
Add some runtime safety checks

### PR DESCRIPTION
**Pull request by gr0uch**
**What kind of change does this PR introduce?**
I ran into some errors within webpack while upgrading a project from v4 to v5, most likely due to its configuration options. Adding some safety checks resolves the errors.

**Did you add tests for your changes?**
Nope :)

**Does this PR introduce a breaking change?**
No.

**What needs to be documented once your changes are merged?**
No documentation required

### CODE REVIEW

**Comment by Cronus1007** [comment]
@gr0uch  What kind of errors you ran into please elaborate?

File: lib/NormalModuleReplacementPlugin.js [hunk 1]
```diff
@@ -43,6 +43,7 @@ class NormalModuleReplacementPlugin {
 				});
 				nmf.hooks.afterResolve.tap("NormalModuleReplacementPlugin", result => {
 					const createData = result.createData;
```
Comment (gr0uch) [inline lib/NormalModuleReplacementPlugin.js:45]: this was undefined, I'm not sure why

File: lib/optimize/MinChunkSizePlugin.js [hunk 2]
```diff
@@ -42,6 +42,7 @@ class MinChunkSizePlugin {
 				},
 				chunks => {
 					const chunkGraph = compilation.chunkGraph;
```
Comment (gr0uch) [inline lib/optimize/MinChunkSizePlugin.js:44]: `chunkGraph` was undefined, I think due to disabling chunking

File: lib/debug/ProfilingPlugin.js [hunk 3]
```diff
@@ -173,7 +173,7 @@ class ProfilingPlugin {
 
 	apply(compiler) {
 		const tracer = createTrace(
-			compiler.intermediateFileSystem,
+			compiler.intermediateFileSystem || require("fs"),
```
Comment (Cronus1007) [inline lib/debug/ProfilingPlugin.js:176]: revert this change

File: lib/NormalModuleReplacementPlugin.js (hunk 1 above)
Comment (Cronus1007) [reply lib/NormalModuleReplacementPlugin.js:45]: Create a reproducible repo depicting these errors.

**Comment by sokra** [comment]
These problems are probably caused by some other problems. Adding safety check would hide these errors, so we don't want to do that
